source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
+        }
+
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    bool build_raster_state()
+    {
+        renderer::commands::CreateRasterStateCommandDesc solid_rs_command_desc;
+        solid_rs_command_desc.fill_mode = renderer::FillMode::SOLID;
+        solid_rs_command_desc.cull_mode = renderer::CullMode::BACK;
+        g_regina_ctx.solid_raster_state = renderer::create_raster_state(rsl::move(solid_rs_command_desc));
+
+        renderer::commands::CreateRasterStateCommandDesc wire_rs_command_desc;
+        wire_rs_command_desc.fill_mode = renderer::FillMode::WIREFRAME;
+        wire_rs_command_desc.cull_mode = renderer::CullMode::BACK;
+        g_regina_ctx.wire_raster_state = renderer::create_raster_state(rsl::move(wire_rs_command_desc));
+
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    bool build_pipeline_state_object()
+    {
+        renderer::commands::CreatePipelineStateCommandDesc create_pso_command_desc;
+        create_pso_command_desc.input_layout = g_regina_ctx.input_layout;
+        create_pso_command_desc.num_render_targets = 1;
+        create_pso_command_desc.shader_program = g_regina_ctx.shader_program;
+#if RENDER_WIREFRAME
+        create_pso_command_desc.rasterizer_state = g_regina_ctx.wire_raster_state;
+#else
+        create_pso_command_desc.rasterizer_state = g_regina_ctx.solid_raster_state;
+#endif
+        g_regina_ctx.pso = renderer::create_pipeline_state_object(rsl::move(create_pso_command_desc));
+
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    void update_view()
+    {
+        // Convert Spherical to Cartesian coordinates.
+        g_regina_ctx.eye_pos.x = 15.0f * sinf(0.2f * glm::pi<f32>()) * cosf(1.5f * glm::pi<f32>());
+        g_regina_ctx.eye_pos.y = 15.0f * cosf(0.2f * glm::pi<f32>());
+        g_regina_ctx.eye_pos.z = 35.0f * sinf(0.2f * glm::pi<f32>()) * sinf(1.5f * glm::pi<f32>());
+
+        // Build the view matrix.
+        glm::vec3 pos = glm::vec3(g_regina_ctx.eye_pos.x, g_regina_ctx.eye_pos.y, g_regina_ctx.eye_pos.z);
+        glm::vec3 target = glm::vec3(0.0f, 0.0f, 0.0f);
+        glm::vec3 up = glm::vec3(0.0f, 1.0f, 0.0f);
+
+        g_regina_ctx.view = glm::lookAt(pos, target, up);
+        g_regina_ctx.view = glm::transpose(g_regina_ctx.view); // DirectX backend ( so we have to transpose, expects row major matrices )
+    }
+    //-------------------------------------------------------------------------
+    void update_object_constant_buffers()
+    {
+        for (auto& ri : (*g_regina_ctx.scene))
+        {
+            // Only update the cbuffer data if the constants have changed.  
+            // This needs to be tracked per frame resource.
+            if (ri.num_frames_dirty > 0)
```

Comment:
this always triggers, but our frames are static, so if I have to believe the comment, this should never trigger

Reply:
This is a leftover from when I had a rotating cube, I removed the line where frames are always marked " dirty "  again.